### DISCUSSION TITLE
Importing Functions, tools and models not working

### DISCUSSION THREAD

**Discussion by Channel1341** [open-webui/open-webui#7250] [General]
# Bug Report

## Important Notes

- **Before submitting a bug report**: Please check the Issues or Discussions section to see if a similar issue or feature request has already been posted. It's likely we're already tracking it! If you’re unsure, start a discussion post first. This will help us efficiently focus on improving the project.

- **Collaborate respectfully**: We value a constructive attitude, so please be mindful of your communication. If negativity is part of your approach, our capacity to engage may be limited. We’re here to help if you’re open to learning and communicating positively. Remember, Open WebUI is a volunteer-driven project managed by a single maintainer and supported by contributors who also have full-time jobs. We appreciate your time and ask that you respect ours.

- **Contributing**: If you encounter an issue, we highly encourage you to submit a pull request or fork the project. We actively work to prevent contributor burnout to maintain the quality and continuity of Open WebUI.

- **Bug reproducibility**: If a bug cannot be reproduced with a `:main` or `:dev` Docker setup, or a pip install with Python 3.11, it may require additional help from the community. In such cases, we will move it to the "issues" Discussions section due to our limited resources. We encourage the community to assist with these issues. Remember, it’s not that the issue doesn’t exist; we need your help!

Note: Please remove the notes above when submitting your post. Thank you for your understanding and support!

---

## Installation Method

Unraid Docker

## Environment

- **Open WebUI Version:** [v0.4.3]
- **Ollama (if applicable):** [v0.4.3]

- **Operating System:** [Unraid]
- **Browser (if applicable):** [Chrome , Edge and Firefox]

**Confirmation:**

- [ yes] I have read and followed all the instructions provided in the README.md.
- [ yes] I am on the latest version of both Open WebUI and Ollama.
- [no ] I have included the browser console logs.
- [no] I have included the Docker container logs.
- [ yes] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

ability to import functions, tools and models

## Actual Behavior:

Manual import doesnt work and web import generates an error

## Description

**Bug Summary:**
Manual import doesnt work, nothing happens at all, and web import generates an error "{ERROR: cannot import name 'PipelineException' from 'transformes' (usr/local/lib/python3.11/site-packages/transformes/_init_py)}"

## Reproduction Details

**Steps to Reproduce:**
open webui, navigate to settings, functions, try to import manually or through web import

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots/Screen Recordings (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by ostapiuk** [comment]
I'm having a similar problem. 

### Environment

**Installation Method:** pip
**Open WebUI Version:** 0.4.8
**Ollama (if applicable):**
**Operating System:** Ubuntu 24.04 container on Proxmox 8.3.1
**Browser:** Brave 1.73.91 (Brave Shields are disabled), Zen 1.0.2-b.2 based on Fireffox 133.0.3
**Python Version:** 3.11

I run Open WebUI as a service using the following startup script:

```bash
#!/bin/bash
source /opt/open-webui/venv/bin/activate
open-webui serve --port 80
```

I am not using a reverse proxy as this is just for personal use on my home LAN.

Functions I attempted to import:
* https://openwebui.com/f/pietrusky/save_outputs
* https://openwebui.com/f/justinrahb/perplexity
* https://openwebui.com/f/iqballx/visualize_v2

### Bug Summary

When I attempt to manually import a function via json file on the /admin/functions page, the process starts normally, but after hitting the 'Confirm' button on the warning dialog, nothing happens.  The result is the same for each function I have tried to import.  After each attempted import, there is an error in the JavaScript console and the WebUI locks up (fixed by reloading):

![console](https://github.com/user-attachments/assets/7a11aec6-20ab-4b00-b945-faea213531d7)

```JavaScript
+layout.svelte:104 Backend config: {status: true, name: 'Open WebUI', version: '0.4.8', default_locale: '', oauth: {…}, …}
+layout.svelte:58 connected LxL-LuyuG3PgAcR7AACB
+layout.svelte:77 user-count {count: 1}
+layout.svelte:82 usage {models: Array(0)}
Functions.svelte:388 FileList {0: File, length: 1}
Functions.svelte:509 [{…}]
index.ts:6 
        
        
       POST http://chat.mydomain.lan/api/v1/functions/create 422 (Unprocessable Entity)
window.fetch @ fetcher.js:76
s @ index.ts:6
_.onload @ Functions.svelte:512
load
ge @ Functions.svelte:507
(anonymous) @ lifecycle.js:105
(anonymous) @ lifecycle.js:104
_ @ ConfirmDialog.svelte:42
await in _
m @ ConfirmDialog.svelte:121
index.ts:23 {detail: Array(3)}
dom.js:1228 Uncaught (in promise) TypeError: t is not a constructor
    at se (dom.js:1228:9)
    at Array.ii (Toast.svelte:338:32)
    at Tt (Toast.svelte:335:42)
    at Array.Qn (Toast.svelte:330:20)
    at li (Toast.svelte:304:22)
    at jt (Component.js:148:34)
    at new di (Toast.svelte:179:5)
    at Ht (Toaster.svelte:216:40)
    at zt (Toaster.svelte:194:11)
    at Ut (Toaster.svelte:163:9)
se @ dom.js:1228
ii @ Toast.svelte:338
Tt @ Toast.svelte:335
Qn @ Toast.svelte:330
li @ Toast.svelte:304
jt @ Component.js:148
di @ Toast.svelte:179
Ht @ Toaster.svelte:216
zt @ Toaster.svelte:194
Ut @ Toaster.svelte:163
p @ Toaster.svelte:161
dt @ scheduler.js:119
_t @ scheduler.js:79
Promise.then
ut @ scheduler.js:20
ht @ Component.js:81
(anonymous) @ Component.js:139
(anonymous) @ Toaster.svelte:54
o @ index.js:56
f @ index.js:69
r @ helpers.js:24
p @ state.js:8
r @ state.js:41
T @ state.js:65
(anonymous) @ Functions.svelte:513
Promise.catch
_.onload @ Functions.svelte:512
load
ge @ Functions.svelte:507
(anonymous) @ lifecycle.js:105
(anonymous) @ lifecycle.js:104
_ @ ConfirmDialog.svelte:42
await in _
m @ ConfirmDialog.svelte:121

```
